### PR TITLE
Added animedit avatar-animation.json graph editor tool

### PR DESCRIPTION
**Pull request by hyperlogic**
animedit is a developer tool used to inspect and edit avatar animation graphs.

NOTE: This does not effect the High Fidelity product or build in anyway.  It is a developer tool that must be built separately from the main product.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22678/HighFidelity-Beta-PR16532-320095f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22678/HighFidelity-Quest-Beta-PR16532-320095f.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22678/HighFidelity-Beta-PR16532-320095f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22678/HighFidelity-Beta-Interface-PR16532-320095f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22678/HighFidelity-Beta-PR16532-320095f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22678/HighFidelity-Beta-Interface-PR16532-320095f.exe

File: tools/animedit/treemodel.cpp [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+    _roleNameMapping[TreeModelRoleData] = "data";
+
+    QList<QVariant> rootData;
+    rootData << "Name" << "Type" << "Data";
+    _rootItem = new TreeItem(rootData);
+    _clipboard = nullptr;
+}
+
+TreeModel::~TreeModel() {
+    delete _rootItem;
```
[the remark this answers is not in the document]
Comment (hyperlogic) [reply tools/animedit/treemodel.cpp]: At the moment the TreeModel is basically a singleton.  There is only one global model created in main.cpp.

File: tools/animedit/deploy.bat [hunk 2]
```diff
@@ -0,0 +1,3 @@
+mkdir C:\Users\ajthy\Documents\animedit\bin
```
[the remark this answers is not in the document]
Comment (hyperlogic) [reply tools/animedit/deploy.bat]: This was a helper file, I shouldn't have check it in.

File: tools/animedit/main.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    // expose model
+    TreeModel model;
+    engine.rootContext()->setContextProperty("theModel", &model);
+
+    const QUrl url(QStringLiteral("qrc:/qml/main.qml"));
+    QObject::connect(&engine, &QQmlApplicationEngine::objectCreated, &app, [url](QObject *obj, const QUrl &objUrl) {
+        if (!obj && url == objUrl) {
+            QCoreApplication::exit(-1);
+        } else {
+            ;
```
[the remark this answers is not in the document]
Comment (hyperlogic) [reply tools/animedit/main.cpp]: I removed the else

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22708/HighFidelity-Beta-PR16532-764b6a9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22708/HighFidelity-Beta-Interface-PR16532-764b6a9.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22708/HighFidelity-Beta-PR16532-764b6a9.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22708/HighFidelity-Quest-Beta-PR16532-764b6a9.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22708/HighFidelity-Beta-PR16532-764b6a9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22708/HighFidelity-Beta-Interface-PR16532-764b6a9.exe

**Pull request merged**